.github/workflows/test-pip.yml

```
@@ -0,0 +1,125 @@
+      GEOS_INSTALL: ${{ github.workspace }}/geosinstall/geos-${{ matrix.geos }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Cache GEOS and pip packages
+        uses: actions/cache@v2
+        with:
+          key: ${{ matrix.os }}-geos-${{ matrix.geos }}
+          path: |
+            ~/.cache/pip
+            ${{ github.workspace }}/geosinstall
+
+      - name: Activate MSVC 64-bit (Windows)
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x64
+        if: ${{ matrix.os == 'windows-2019' }}
+
+      - name: Correct slashes in GEOS_INSTALL (Windows)
+        shell: bash
+        run: |
+          echo 'GEOS_INSTALL=${{ github.workspace }}\geosinstall\geos-${{ matrix.geos }}' >> $GITHUB_ENV
+        if: ${{ matrix.os == 'windows-2019' }}
+
+      - name: Install GEOS
+        shell: bash
+        run: |
+          ./ci/install_geos.sh
+          cd '${{ github.workspace }}'
+
+      - name: Install python dependencies
+        shell: bash
+        run: |
+          pip install --disable-pip-version-check --upgrade pip
+          pip install --upgrade wheel setuptools
+          if [ "$GEOS_VERSION" = "main" ]; then
+            pip install --upgrade --pre Cython numpy pytest;
+          else
+            pip install --upgrade Cython numpy==${{ matrix.numpy }} pytest;
+          fi
+          pip list
+
+      - name: Set environment variables (Linux)
+        shell: bash
+        run: |
+          echo "${{ env.GEOS_INSTALL }}/bin" >> $GITHUB_PATH
+          echo "LD_LIBRARY_PATH=${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+
+      - name: Set environment variables (OSX)
+        shell: bash
+        run: |
+          echo "${{ env.GEOS_INSTALL }}/bin" >> $GITHUB_PATH
+          echo "DYLD_FALLBACK_LIBRARY_PATH=${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+          echo "LDFLAGS=-Wl,-rpath,${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+        if: ${{ matrix.os == 'macos-latest' }}
+
+      - name: Set environment variables (Windows)
+        shell: bash
+        run: |
+          cp geosinstall/geos-${{ matrix.geos }}/bin/*.dll pygeos
+          echo 'GEOS_LIBRARY_PATH=${{ env.GEOS_INSTALL }}\lib' >> $GITHUB_ENV
+          echo 'GEOS_INCLUDE_PATH=${{ env.GEOS_INSTALL }}\include' >> $GITHUB_ENV
+        if: ${{ matrix.os == 'windows-2019' }}
+
+      - name: Build PyGEOS
+        shell: bash
+        run: |
+          python setup.py build_ext --inplace
+          pip install --no-deps -e .
+
+      - name: Run tests
+        shell: bash
+        continue-on-error: ${{ matrix.geos == 'main' }}
+        run: |
+          pytest pygeos
+
+      - name: Run doctests
+        shell: bash
+        run: |
+          rm -rf pygeos/tests 
+          pytest --doctest-modules pygeos
```

```suggestion
          pytest --doctest-modules pygeos --ignore=pygeos/tests
```

(ignores test directory instead of needing to delete them)